src/commands/mod.rs

```diff
@@ -120,18 +120,23 @@ fn configure_args(config: &mut Config, matches: &ArgMatches) -> Result<()> {
         config.set_headers(headers);
     }
 
-    if let Some(level_str) = matches.get_one::<String>("log_level") {
-        match level_str.parse() {
-            Ok(level) => {
-                config.set_log_level(level);
+    Ok(())
+}
+
+pub fn configure_log_level(matches: &ArgMatches) -> Result<Option<LevelFilter>> {
+    match matches.get_one::<String>("log_level") {
+        Some(log_level) => match log_level.parse() {
+            Ok(log_level) => {
+                set_max_level(log_level);
```

Comment:
I would prefer this to happen at the call site instead of in this function. I think it's very easy to miss here.